tests/blob/test_block_blob.py

```diff
@@ -520,6 +520,55 @@ def test_create_blob_from_path_with_properties(self):
         self.assertEqual(properties.content_settings.content_type, content_settings.content_type)
         self.assertEqual(properties.content_settings.content_language, content_settings.content_language)
 
+    def test_createBlobFromStream_when_short_read_the_whole_stream(self):
+        # to get the actual uploaded blob, can only run live
+        if TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Arrange
+        blob_name = self._get_blob_reference()
+        content = b"Example blob content."
+        stream = ShortReader(io.BytesIO(content), 10)
+        upload_size = len(content)
+
+        # Act
+        self.bs.create_blob_from_stream(self.container_name, blob_name, stream, upload_size)
+
+        # Assert
+        uploaded_blob = self.bs.get_blob_to_bytes(self.container_name, blob_name)
+        self.assertEquals(content, uploaded_blob.content)
+
+    def test_createBlobFromStream_when_short_read_designated_size_of_stream(self):
+        # to get the actual uploaded blob, can only run live
+        if TestMode.need_recording_file(self.test_mode):
+            return
+
+        # Arrange
+        blob_name = self._get_blob_reference()
+        content = b"Example blob content."
+        stream = ShortReader(io.BytesIO(content), 10)
+        upload_size = 15  # designated size blob we want to upload
+
+        # Act
+        self.bs.create_blob_from_stream(self.container_name, blob_name, stream, upload_size)
+
+        # Assert
+        uploaded_blob = self.bs.get_blob_to_bytes(self.container_name, blob_name)
+        self.assertEquals(upload_size, len(uploaded_blob.content))
+        self.assertEquals(content[:upload_size], uploaded_blob.content)
+
+    def test_createBlobFromStream_when_designated_size_larger_than_the_actual_stream_size(self):
```

Comment:
we shouldn't throw error when there's more data in the stream. please remove this test

Reply:
this test is for when the stream doesn't have enough data, which is len(stream) < user specified